File: src/main/java/core/listeners/LeaverListener.java
package core.listeners;

import db.model.Member;
import net.dv8tion.jda.api.events.guild.member.GuildMemberLeaveEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import utils.BotUtils;
import utils.DBUtils;
import utils.MessageUtils;

public class LeaverListener extends ListenerAdapter {

    @Override
    public void onGuildMemberLeave(GuildMemberLeaveEvent event) {
        Member member = new Member(event.getMember().getUser().getIdLong(), event.getGuild().getIdLong());
        DBUtils.delete(member);
        if (event.getGuild().getIdLong() == BotUtils.SRID) {
            MessageUtils.sendLog(event.getUser().getName() + " has left");
        }
    }
}

File: src/main/kotlin/core/listeners/LeaverListener.kt
package core.listeners

import internal.models.Member
import net.dv8tion.jda.api.events.guild.member.GuildMemberLeaveEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter
import utils.BotUtils
import utils.DBUtils
import utils.MessageUtils

class LeaverListener : ListenerAdapter() {

    override fun onGuildMemberLeave(event: GuildMemberLeaveEvent) {
        val member = Member(event.member.user.idLong, event.guild.idLong)
        DBUtils.delete(member)
        if (event.guild.idLong == BotUtils.SRID) {
            MessageUtils.sendLog(event.user.name + " has left")
        }
    }
}
